### PR TITLE
Force jupyter-server v1 to check against notebook v6

### LINKED ISSUE #13563: 3.6 branch CI is failing since jupyter-server 2.0 release
> I don't know what to think about the CI failures. They seems unrelated and they happen also on the latest merge on https://github.com/jupyterlab/jupyterlab/pull/13543. But I'm surprised by their failure on the second attempt.

by @fcollonval in https://github.com/jupyterlab/jupyterlab/pull/13555#pullrequestreview-1210406165

> I think this might have something to do with Jupyter Server 2.0 release two days ago. I see that the failures are in server-related tests and that jupyter-server-2.0.0 does get installed, so looks pretty solid.
>
> I re-run Linux (usage, 3.10) with naive hope that 2.0.1 that was released 10 hours ago may have fixed it, but it does not.
>
> Maybe some server 2.0 compatibility thing that was not backported to 3.6.x?

by @krassowski in https://github.com/jupyterlab/jupyterlab/pull/13555#issuecomment-1343461554

One of the offenders appears to be a browser check using notebook server:

https://github.com/jupyterlab/jupyterlab/blob/f209231f9f3790645e5b09f61f4a1fa2adec4fb3/scripts/ci_script.sh#L240-L254

whereas on master it passes because... there isn't one:

https://github.com/jupyterlab/jupyterlab/blob/e658b6f28e3501968e7fe8e211677c5378be14d0/scripts/ci_script.sh#L234-L239

<details>

The relevant log is:

```python
[W 09:34:39.685 NotebookApp] Config option `nbserver_extensions` not recognized by `NotebookApp`.  Did you mean one of: `reraise_server_extension_failures, server_extensions`?
[I 09:34:39.698 NotebookApp] Writing notebook server cookie secret to /Users/runner/Library/Jupyter/runtime/notebook_cookie_secret
[I 2022-12-08 09:34:39.881 LabApp] JupyterLab extension loaded from /Users/runner/work/jupyterlab/jupyterlab/jupyterlab
[I 2022-12-08 09:34:39.881 LabApp] JupyterLab application directory is /Users/runner/hostedtoolcache/Python/3.10.8/x64/share/jupyter/lab
[I 2022-12-08 09:34:39.912 BrowserApp] JupyterLab extension loaded from /Users/runner/work/jupyterlab/jupyterlab/jupyterlab
[I 2022-12-08 09:34:39.912 BrowserApp] JupyterLab application directory is /Users/runner/hostedtoolcache/Python/3.10.8/x64/share/jupyter/lab
[I 09:34:39.918 NotebookApp] Serving notebooks from local directory: /Users/runner/work/jupyterlab/jupyterlab
[I 09:34:39.919 NotebookApp] Jupyter Notebook 6.5.2 is running at:
[I 09:34:39.919 NotebookApp] http://localhost:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.919 NotebookApp]  or http://127.0.0.1:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.919 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 09:34:39.926 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///Users/runner/Library/Jupyter/runtime/nbserver-43188-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
     or http://127.0.0.1:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.931 NotebookApp] Running async test
Starting headless webkit...
Navigating to page: file:///Users/runner/Library/Jupyter/runtime/nbserver-43188-open.html
Waiting for page to load...
/Users/runner/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/jupyter_server/base/handlers.py:232: UserWarning: The Tornado web application does not have an 'identity_provider' defined in its settings. In future releases of jupyter_server, this will be a required key for all subclasses of `JupyterHandler`. For an example, see the jupyter_server source code for how to add an identity provider to the tornado settings: https://github.com/jupyter-server/jupyter_server/blob/aa8fd8b3faf37466eeb99689d5555314c5bf6640/jupyter_server/serverapp.py#L253
  warnings.warn(
[I 09:34:43.098 NotebookApp] 302 GET /lab?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8 (::1) 4.140000ms
Waiting for page content..
Error loading JupyterLab page:
Waiting for #main selector...
<!DOCTYPE html><html lang="en-us"><head>
    <meta charset="utf-8">

    <title>Jupyter Notebook</title>
```

</details>

It was removed in commit 5a95a5575a0353af88ba29de984d17357f2e6b2a and PR #11894.

- should we backport #11894?
- should we test 3.6 with both jupyter-server 1.x and 2.x?

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fix #13563

## Code changes

Test the version of `jupyter_server` when loading JupyterLab as an old notebook extension.
```
If `jupyter_server` version >= 2.0.0:
    if default url == '/lab':
       sys.exit
    else:
       Skip adding JupyterLab as extension
```

## User-facing changes

Users are not able to access JupyterLab from Classic notebook except if jupyter server v1 is installed.

## Backwards-incompatible changes

None - in the sense that the current situation is: user will face a login page to access JupyterLab that will reject all entries.

### CODE REVIEW

File: jupyterlab/serverextension.py [hunk 1]
```diff
@@ -8,6 +10,22 @@ def load_jupyter_server_extension(serverapp):
     """Temporary server extension shim when using
     old notebook server.
     """
+    serverapp.log.warning("Loading JupyterLab as an old notebook extension.")
```
Comment (jtpio) [inline jupyterlab/serverextension.py:13]: Maybe something more explicit like the following?

```suggestion
    serverapp.log.warning("Loading BrowserApp as a classic notebook (v6) extension.")
```

Comment (jtpio) [reply jupyterlab/serverextension.py:13]: To disambiguate the word "old".

Comment (fcollonval) [reply jupyterlab/serverextension.py:13]: Thanks - just a comment this is not the `BrowserApp`, this is `JupyterLab` server extension hook.

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.5.x

**Comment by jtpio** [comment]
cc @echarles @RRosio @ericsnekbytes for awareness

**Pull request merged**